src/coreclr/jit/objectalloc.cpp

```diff
@@ -1673,33 +1720,105 @@ bool ObjectAllocator::CanLclVarEscapeViaParentStack(ArrayStack<GenTree*>* parent
             case GT_STOREIND:
             case GT_STORE_BLK:
             case GT_BLK:
-                if (tree != parent->AsIndir()->Addr())
+            {
+                GenTree* const addr = parent->AsIndir()->Addr();
+                if (tree == addr)
+                {
+                    JITDUMP("... store address\n");
+                    canLclVarEscapeViaParentStack = false;
+                    break;
+                }
+
+                // If the value being stored is a local address, anything assigned to that local escapes
+                //
+                if (isAddress)
                 {
-                    // TODO-ObjectStackAllocation: track stores to fields.
                     break;
                 }
-                FALLTHROUGH;
+
+                // Is this a store to a field of a local struct...?
+                //
+                if (parent->OperIs(GT_STOREIND))
+                {
+                    // Are we storing to a local field?
+                    //
+                    if (addr->OperIs(GT_FIELD_ADDR))
+                    {
+                        // Simple check for which local.
+                        //
+                        GenTree* const base = addr->AsOp()->gtGetOp1();
+
+                        if (base->OperIs(GT_LCL_ADDR))
+                        {
+                            unsigned const   dstLclNum = base->AsLclVarCommon()->GetLclNum();
+                            LclVarDsc* const dstDsc    = comp->lvaGetDesc(dstLclNum);
+
+                            if (IsTrackedLocal(dstLclNum))
+                            {
+                                JITDUMP("... local.field store\n");
+                                // Add an edge to the connection graph.
+                                AddConnGraphEdge(dstLclNum, lclNum);
+                                canLclVarEscapeViaParentStack = false;
+                            }
+                        }
+                    }
```

Comment:
I thought you were going with the more conservative approach, but evidently not completely?